### PR TITLE
Update vertx dependencies

### PR DESCRIPTION
**Pull request by OwenCorrigan76**
Update Vertx dependencies from version 3.7.1 to version 4.3.7.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 RELEASE_VERSION ?= latest
 
 SUBDIRS=java/common java/kafka/consumer java/kafka/producer java/kafka/streams java/http/java-http-consumer java/http/java-http-producer java/http/vertx/java-http-vertx-consumer java/http/vertx/java-http-vertx-producer
+
```
Comment (kyguy) [inline Makefile]: extra line

File: java/http/deployment-tracing-opentelemetry.yaml [hunk 2]
```diff
@@ -26,6 +26,8 @@ spec:
             value: my-topic
           - name: STRIMZI_SEND_INTERVAL
             value: "1000"
+          - name: STRIMZI_MESSAGE_COUNT
+            value: "1000000"
```
Comment (kyguy) [inline java/http/deployment-tracing-opentelemetry.yaml]: Stale

File: java/http/deployment-tracing-opentelemetry.yaml [hunk 3]
```diff
@@ -70,6 +72,8 @@ spec:
             value: "my-group"
           - name: STRIMZI_POLL_INTERVAL
             value: "1000"
+          - name: STRIMZI_MESSAGE_COUNT
+            value: "1000000"
```
Comment (kyguy) [inline java/http/deployment-tracing-opentelemetry.yaml]: Stale

File: java/http/java-http-consumer.yaml [hunk 4]
```diff
@@ -30,6 +30,8 @@ spec:
             value: "my-group"
           - name: STRIMZI_POLL_INTERVAL
             value: "1000"
+          - name: STRIMZI_MESSAGE_COUNT
+            value: "1000000"
```
Comment (kyguy) [inline java/http/java-http-consumer.yaml]: Stale

File: java/http/java-http-producer.yaml [hunk 5]
```diff
@@ -26,6 +26,8 @@ spec:
             value: my-topic
           - name: STRIMZI_SEND_INTERVAL
             value: "1000"
+          - name: STRIMZI_MESSAGE_COUNT
+            value: "1000000"
```
Comment (kyguy) [inline java/http/java-http-producer.yaml]: Stale

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java [hunk 6]
```diff
@@ -9,23 +9,22 @@
...
 import io.opentracing.propagation.TextMapAdapter;
 import io.opentracing.tag.Tags;
 import io.opentracing.util.GlobalTracer;
-import io.vertx.core.MultiMap;
-
-import io.netty.handler.codec.http.HttpHeaderNames;
-import io.netty.handler.codec.http.HttpResponseStatus;
-import io.vertx.core.AbstractVerticle;
-import io.vertx.core.Future;
+import io.vertx.core.*;
```
Comment (kyguy) [inline java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: My cause checkstyle errors, avoid wildcard imports and import each lib individual

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java [hunk 7]
```diff
@@ -49,7 +48,7 @@ public class HttpKafkaConsumer extends AbstractVerticle {
 
     /**
      * Constructor
-     * 
+     *
```
Comment (kyguy) [inline java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java:53]: extra line

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java [hunk 8]
```diff
@@ -68,190 +67,192 @@ public void start(Future<Void> startFuture) throws Exception {
                 .setPipelining(this.config.isPipelining())
                 .setPipeliningLimit(this.config.getPipeliningLimit());
         this.client = WebClient.create(vertx, options);
-        
+
```
Comment (kyguy) [inline java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java:72]: extra line

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java [hunk 9]
```diff
@@ -68,190 +67,192 @@ public void start(Future<Void> startFuture) throws Exception {
...
+    }
+
     private Future<CreatedConsumer> createConsumer() {
-        Future<CreatedConsumer> fut = Future.future();
+        Promise<CreatedConsumer> promise = Promise.promise();
 
         JsonObject json = new JsonObject()
-            .put("format", "json");
-        
+                .put("format", "json");
```
Comment (kyguy) [inline java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: extra spaces

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java [hunk 10]
```diff
@@ -68,190 +67,192 @@ public void start(Future<Void> startFuture) throws Exception {
...
-            .putHeader(HttpHeaderNames.CONTENT_TYPE.toString(), "application/vnd.kafka.v2+json")
-            .as(BodyCodec.jsonObject())
-            .sendJsonObject(json, ar -> {
-                if (ar.succeeded()) {
-                    HttpResponse<JsonObject> response = ar.result();
+                .putHeader(HttpHeaderNames.CONTENT_LENGTH.toString(), String.valueOf(json.toBuffer().length()))
+                .putHeader(HttpHeaderNames.CONTENT_TYPE.toString(), "application/vnd.kafka.v2+json")
+                .as(BodyCodec.jsonObject())
+                .sendJsonObject(json)
+                .onSuccess(response -> {
```
Comment (kyguy) [inline java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: undo indentation changes

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java [hunk 11]
```diff
@@ -68,190 +67,192 @@ public void start(Future<Void> startFuture) throws Exception {
...
-                        MultiMap rawHeaders = response.headers();
-                        final Map<String, String> headers = new HashMap<>();
-                        for (Map.Entry<String, String> header : rawHeaders) {
-                            headers.put(header.getKey(), header.getValue());
-                        }
+                            MultiMap rawHeaders = response.headers();
+                            final Map<String, String> headers = new HashMap<>();
+                            for (Map.Entry<String, String> header : rawHeaders) {
+                                headers.put(header.getKey(), header.getValue());
+                            }
```
Comment (kyguy) [inline java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: Fix indentation

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java [hunk 12]
```diff
@@ -68,190 +67,192 @@ public void start(Future<Void> startFuture) throws Exception {
...
+                                        json.getInteger("partition"),
+                                        json.getLong("offset"))
                                 );
-                        });
-                        this.messagesReceived += list.size();
+                            });
+                            this.messagesReceived += list.size();
 
-                        span.finish();
+                            span.finish();
```
Comment (kyguy) [inline java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: fix indentation

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java [hunk 13]
```diff
@@ -68,190 +67,192 @@ public void start(Future<Void> startFuture) throws Exception {
...
+                    if (this.config.getMessageCount().isPresent() &&
+                            this.messagesReceived >= this.config.getMessageCount().get()) {
                         // signal to main thread that all messages are received, application can exit
                         this.messagesReceivedLatch.countDown();
                         log.info("All messages received");
-                }
-            });
-        return fut;
+                    }
+                });
```
Comment (kyguy) [inline java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: fix indentation

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java [hunk 14]
```diff
@@ -68,190 +67,192 @@ public void start(Future<Void> startFuture) throws Exception {
...
         return fut;
     }
 
     /**
      * Information about using the consumer on the bridge
      */
     class CreatedConsumer {
-    
+
```
Comment (kyguy) [inline java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java:254]: extra line

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java [hunk 15]
```diff
@@ -68,190 +67,192 @@ public void start(Future<Void> startFuture) throws Exception {
...
         private final String instanceId;
         private final String baseUri;
-        
+
```
Comment (kyguy) [inline java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java:257]: extra line

File: java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducer.java [hunk 16]
```diff
@@ -15,13 +15,13 @@
 import io.opentracing.propagation.TextMap;
 import io.opentracing.tag.Tags;
 import io.opentracing.util.GlobalTracer;
+import io.vertx.core.Promise;
 import io.vertx.core.buffer.Buffer;
 import io.vertx.ext.web.client.HttpRequest;
+import io.netty.handler.codec.http.*;
+
```
Comment (kyguy) [inline java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducer.java]: extra line

File: Makefile (hunk 1 above)
Comment (OwenCorrigan76) [reply Makefile]: Removed

File: java/http/deployment-tracing-opentelemetry.yaml (hunk 2 above)
Comment (OwenCorrigan76) [reply java/http/deployment-tracing-opentelemetry.yaml]: This was removed on the last PR. Not sure how it got in again. Now removed.

File: java/http/deployment-tracing-opentelemetry.yaml (hunk 3 above)
Comment (OwenCorrigan76) [reply java/http/deployment-tracing-opentelemetry.yaml]: Same as above - now removed.

File: java/http/java-http-consumer.yaml (hunk 4 above)
Comment (OwenCorrigan76) [reply java/http/java-http-consumer.yaml]: Same as above - now removed

File: java/http/java-http-producer.yaml (hunk 5 above)
Comment (OwenCorrigan76) [reply java/http/java-http-producer.yaml]: Same as above - now removed

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 12 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: Fixed

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 13 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: Fixed??

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 14 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java:254]: Indentation on an empty line??

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 15 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java:257]: The empty line was on 254, now on 255

File: java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducer.java (hunk 16 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducer.java]: Removed

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 11 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: Fixed

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 10 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: Fixed (I think)

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 9 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: Fixed

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 8 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java:72]: ?? There's a line space in the original??

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 7 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java:53]: ?? Same as original

File: java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducer.java (hunk 16 above)
Comment (ppatierno) [reply java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducer.java]: We avoid to have `*` when importing classes but listing them all.

**Review by ppatierno — CHANGES_REQUESTED** [review]

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 6 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: Removed wildcard and used individual specific imports

File: java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducer.java (hunk 16 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducer.java]: All wildcard imports removed and replaced with appropriate imports

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java [hunk 17]
```diff
@@ -68,190 +69,192 @@ public void start(Future<Void> startFuture) throws Exception {
...
                         for (Map.Entry<String, String> header : rawHeaders) {
                             headers.put(header.getKey(), header.getValue());
                         }
 
-                        String operation = "poll";
-                        Tracer.SpanBuilder spanBuilder;
-                        try {
-                            SpanContext parentSpan = tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMapAdapter(headers));
-                            if (parentSpan == null) {
+                            String operation = "poll";
```
Comment (scholzj) [inline java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: This seems to be misaligned -> the indentation of line 174 and 180 is different from 175-179.

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java [hunk 18]
```diff
@@ -68,190 +69,192 @@ public void start(Future<Void> startFuture) throws Exception {
...
+                    log.info("Subscribed to {}", topic);
+                    promise.complete();
+                } else {
+                    promise.fail(new RuntimeException("Got HTTP status code " + response.statusCode()));
+                }
+            } else {
+                promise.fail(ar != null ? ar.cause() : new NullPointerException("ar is null"));
+            }
+        });
+        return promise.future();
```
Comment (scholzj) [inline java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: Just a tip: if you leave this method where the original was, the diff will look nicer and reviews will be much easier for reviewers as they would see the old and new code next to each other. There does not seem to be any real reason why to move this method within the same class.

**Review by scholzj — COMMENTED** [review]
Left some nits. But looks mostly good.

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 18 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: Not sure why I moved the subscribe method. Put back to correct place.

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 17 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: I've realigned the whole HttpKafkaConsumer class to where it was in the previous commits. Hopefully this is correct?

**Review by scholzj — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]
LGTM

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java [hunk 19]
```diff
@@ -68,94 +69,96 @@ public void start(Future<Void> startFuture) throws Exception {
...
 
     private Future<Void> subscribe(CreatedConsumer consumer, String topic) {
-        Future<Void> fut = Future.future();
+        Promise<Void> promise = Promise.promise();
 
         JsonObject topics = new JsonObject()
-            .put("topics", new JsonArray().add(topic));
-        
-        this.client.post(consumer.getBaseUri() + "/subscription")
+                .put("topics", new JsonArray().add(topic));
```
Comment (kyguy) [inline java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: Extra indentation

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 7 above)
Comment (kyguy) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java:53]: This is fine, just deleted an extra space, ignore this comment

**Review by kyguy — APPROVED** [review]
Looks great @OwenCorrigan76, well done!

File: java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java (hunk 19 above)
Comment (OwenCorrigan76) [reply java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java]: Fixed

**Comment by scholzj** [comment]
@OwenCorrigan76 We already approved it -> why does it need another review?

**Review by kyguy — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]
Approving again ;-)

**Comment by OwenCorrigan76** [comment]
@ppatierno @scholzj  Sorry re-requested review after you both approved. My bad!

**Review by scholzj — APPROVED** [review]

**Pull request merged**